llvm/lib/Target/Mips/MipsTargetMachine.cpp

```diff
@@ -62,6 +62,8 @@ extern "C" LLVM_EXTERNAL_VISIBILITY void LLVMInitializeMipsTarget() {
   initializeMipsPreLegalizerCombinerPass(*PR);
   initializeNMOptimizeJumpTablesPass(*PR);
   initializeRedundantCopyEliminationPass(*PR);
+  initializeNMLoadStoreOptPass(*PR);
```

Comment:
How did we previously register the pass with pass manager? Do you need to remove that?

Reply:
It was not explicitly registered, which was the problem. It was just created and added in the `MipsPassConfig::addPreSched2()` -- so it was *run*, but otherwise the pass manager didn't know about it.